### PR TITLE
J2735 2024 SPaT Pipeline Improvements

### PR DESCRIPTION
**Pull request by drewjj**
# PR Details
## Description

Includes a complete overhaul of the SPaT pipeline from both UDP and log offload endpoints.
- New J2735 2024 POJOs from the jpo-asn-pojos project have been incorporated into the SPaT pipeline and all output now conforms 100% to the J2735 2024 as defined by the ASN.
- Unused and redundant POJO Kafka topics and classes have been removed.
- The log file offload endpoint records SPaT messages as `rxMsg` types in order to create a differentiation between UDP and log ingested SPaT messages. UDP endpoint maintains the message type as `spatTx`. The rationale here is because all SPaT log files are OBU recorded messages that are received while UDP ingested SPaTs are forwarded by a RSU after transmission.
- Unit tests have been updated to conform to the new data formats and pipeline.
- The flowchart has been updated to include the current state of the SPaT pipeline as well as noting the different record types based on message ingestion endpoint.
- Some logic around distress notification log ingestion has been changed. Preparing to potentially deprecate this message type.

## Related Issue

N/A

## How Has This Been Tested?

This has been tested locally in a single instance environment through the following methods:
- SPaT UDP port 44910 endpoint using the SPaT UDP send script found on the jpo-ode project's [scripts/tests directory](https://github.com/CDOT-CV/jpo-ode/tree/master/scripts/tests).
- Unit tests were utilized to ensure known corner cases were tested properly.

Note: We have no known spat OBU logs to test with. CDOT's OBUs do not support this message type, either. This functionality was maintained since it existed and there is no good reason to remove it.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[ODE Contributing Guide](https://github.com/usdot-jpo-ode/jpo-ode/blob/develop/docs/contributing_guide.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by dmccoystephenson — APPROVED** [review]
Looks good to me!

- The unit tests pass
- SPATs can be ingested via UDP
- JSON in the topic.odeSpatJson topic passes the schema

File: jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/importer/parser/LogFileParser.java [hunk 1]
```diff
@@ -160,19 +158,12 @@ public void updateMetadata(OdeLogMetadata metadata) {
       } else if (this instanceof RxMsgFileParser rxMsgFileParser
           && rxMsgFileParser.getRxSource() == RxSource.RV) {
         source = Source.RV;
+      } else if (this instanceof SpatLogFileParser spatLogFileParser) {
+        source = Source.valueOf(spatLogFileParser.getSpatSource().toString());
+        odeMFMetadata.setCertPresent(spatLogFileParser.isCertPresent());
       }
```
[the remark this answers is not in the document]
Comment (drewjj) [reply jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/importer/parser/LogFileParser.java:177]: Addressed. I ended up keeping the RxMsg RxSource if statement inside the RxMsgFileParser case. The 'when' keyword is not a standard Java thing from the searches I did so I opted to not do that. (It seems to be a Kotlin thing?) The switch statement is still used, though.

[the remark this answers is not in the document]
Comment (drewjj) [reply jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/importer/parser/LogFileParser.java:177]: Interesting!

[the remark this answers is not in the document]
Comment (drewjj) [reply jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/importer/parser/LogFileParser.java:173]: Alright I switched it to use "when".
